### PR TITLE
remove more pre 1.6 code and simplify kwarg passing

### PR DESCRIPTION
**Pull request by KristofferC**
Just a small cleanup that is possible now when we moved to Julia 1.6.

### CODE REVIEW

File: src/PackageCompiler.jl [hunk 1]
```diff
@@ -517,19 +516,19 @@ function create_sysimage(packages::Union{Symbol, Vector{Symbol}}=Symbol[];
     # Create the sysimage
     object_file = tempname() * ".o"
     create_sysimg_object_file(object_file, packages;
-                              project=project,
```
Comment (Sacha0) [inline src/PackageCompiler.jl:520]: Nice, I forgot about this sugar :).

File: src/PackageCompiler.jl [hunk 2]
```diff
@@ -1064,7 +1061,7 @@ function bundle_artifacts(ctx, dest_dir, library_only; include_lazy_artifacts=tr
                         continue
                     end
                     meta = Pkg.Artifacts.artifact_meta(name, artifacts_toml_path)
-                    meta == nothing && continue
+                    meta === nothing && continue
```
Comment (Sacha0) [inline src/PackageCompiler.jl:1064]: Do I remember correctly that `isnothing(meta)` would optimize just as well on 1.7, but not quite yet on 1.6? :)

**Review by Sacha0 — APPROVED** [review]
Nice cleanups! :)

I was on the fence about the kwarg sugar when it went in, but seeing it in action is convincing!

File: src/PackageCompiler.jl (hunk 2 above)
Comment (KristofferC) [reply src/PackageCompiler.jl:1064]: I thin it does, yes.

**Pull request merged**